src/redux/books/books.js

```diff
@@ -1,26 +1,94 @@
 const ADD_BOOK = 'bookstore/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
 
+const apiUrl = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/Gqq4TbGrZmbMnwiy8Tsp/books';
+const GET_BOOKS = 'bookstore/books/GET_BOOKS';
 const initialState = [];
 
-export const addbook = (payload) => ({
-  type: ADD_BOOK,
-  payload,
+const getBooksFromApi = async () => {
+  const response = await fetch(apiUrl, {
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+  const books = await response.json();
+  const booksData = Object.keys(books).map((id) => ({
+    id,
+    title: books[id][0].title,
+    author: books[id][0].author,
+  }));
+  return booksData;
+};
+
+export const getbooks = () => (async (dispatch) => {
+  const booksData = await getBooksFromApi();
+  dispatch({
+    type: GET_BOOKS,
+    payload: booksData,
+  });
+});
```

Comment:
- Good job on the job done so far 👍 👍. Though your app works, there are improvements to be made on your code, please create an action to be dispatched when books are fetched from the api.
- Instead of having two functions to fetch the books from the API, I suggest that the `getBooksFromApi` function only should be used, the action to fetch the books can be dispatched in the `getBooksFromApi` function, this would help make your code short and easy to maintain, please fix this 🙏